src/vm/amd64/asmhelpers.S

```diff
@@ -0,0 +1,175 @@
+//       Not Preserved:
+//           - integer argument registers (rcx, rdx, r8, r9)
+//           - floating point argument registers (xmm1-3)
+//           - volatile integer registers (r10, r11)
+//           - volatile floating point registers (xmm4-5)
+//           - upper halves of ymm registers on AVX (which are volatile)
+//
+// ***********************************************************
+
+// EXTERN_C void ProfileEnterNaked(FunctionIDOrClientID functionIDOrClientID, size_t profiledRsp);
+// <NOTE>
+//
+// </NOTE>
+NESTED_ENTRY ProfileEnterNaked, _TEXT, NoHandler
+  //       Upon entry :
+  //           rdi = clientInfo
+  //           rsi = profiledRsp
+
+  push_nonvol_reg         rax
+
+  lea                     rax, [rsp + 0x10]    // caller rsp
+  mov                     r10, [rax - 0x8]     // return address
+
+  alloc_stack             SIZEOF_STACK_FRAME
+
+  // correctness of return value in structure doesn't matter for enter probe
+
+  // setup ProfilePlatformSpecificData structure
+  xor                     r11, r11 // nullify r11
+  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA +  0x0], r11    // r11 is null     -- struct functionId field
+  save_reg_postrsp        rbp, OFFSETOF_PLATFORM_SPECIFIC_DATA +    0x8          //                 -- struct rbp field
+  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x10], rax    // caller rsp      -- struct probeRsp field
+  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x18], r10    // return address  -- struct ip field
+  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x20], rsi    //                 -- struct profiledRsp field
+  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x28], r11    // return value    -- struct rax field
+  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x30], r11    // r11 is null     -- struct hiddenArg field
+  movsd                   real8 ptr [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x38], xmm0    //      -- struct flt0 field
+  movsd                   real8 ptr [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x40], xmm1    //      -- struct flt1 field
+  movsd                   real8 ptr [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x48], xmm2    //      -- struct flt2 field
+  movsd                   real8 ptr [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x50], xmm3    //      -- struct flt3 field
+  mov                     r10, 0x1 // PROFILE_ENTER
+  mov                     [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA + 0x58], r10d   // flags    ;      -- struct flags field
+
+  // we need to be able to restore the fp return register
+  save_xmm128_postrsp     xmm0, OFFSETOF_FP_ARG_SPILL + 0x0
+  END_PROLOGUE
+
+  // rdi already contains the clientInfo
+  lea                     rsi, [rsp + OFFSETOF_PLATFORM_SPECIFIC_DATA]
+  call                    ProfileEnter
+
+  // restore fp return register
+  movdqa                  xmm0, [rsp + OFFSETOF_FP_ARG_SPILL + 0x0]
```

Comment:
On Unix AMD64, xmm1 can be used for return value as well (in case of returning a struct). So we need to save / restore it too (in both functions)